tasks/Module 4/task_2.py

```diff
@@ -0,0 +1,31 @@
+"""Fill in the Gaps."""
+import os
+
+
+def fill_gaps(folder_path, prefix):
+    """Fill in the Gaps."""
+    # init file list with first not needed element
+    file_list = [None]
+    os.chdir(folder_path)
+    # get list of tuples of filenames and extentions
+    with os.scandir(folder_path) as scandir:
+        for entry in scandir:
+            if entry.name.startswith(prefix) and entry.is_file():
+                file_list.append(os.path.splitext(entry.name))
+    # check if sequence number equal to number in filename and rename it if is not.
+    number_of_zeroes = len(file_list[1][0]) - len(prefix)
+    for i in range(1, len(file_list)):
```

Comment:
Usually there is no need for writing `range(len(...))` kind of loops in Python. If you are interested in both item and its index you can use `enumerate`:
```python
for i, file_name in enumerate(file_list, start=1):
    ...
```

Reply:
agree, fixed.